### DISCUSSION TITLE
Android Studio - How to select Java as the language

### DISCUSSION THREAD

**Discussion by ahmedmuhmmed8080** [community/community#131951] [Programming Help]
### Select Topic Area

Question

### Body

I learned java, and i wanted to build an Android application, therefore, i downloaded Android Studio. But unfortunately i encountered a problem, and it is briefly that the java language option did not appear. I had only one default option which was kotlin. However, manually, i tried to convert the kotlin files into java files. But no matter how i tried it did not work, I also tried to delete and  download android studio again but
![Screenshot 2024-07-12 161934](https://github.com/user-attachments/assets/336f8815-d131-484e-ad9d-99e1d8a59bc1)
 the exact same problem remained. How can i solve this issue?

**Comment by paiineli** [comment]
Hi,

It sounds like you're running into some frustrating issues with Android Studio not showing the Java language option. Here are a few steps you can take to resolve this:

Create a New Project: When starting a new project in Android Studio, make sure to select the option for Java when prompted about the project type. If you don’t see it, you might need to manually configure it in the project settings.

Check SDK and JDK Installation: Ensure that you have the correct Java Development Kit (JDK) installed. You can check this in Android Studio under File > Project Structure > SDK Location. Make sure the JDK path is set correctly.

Configure Gradle: In your project’s build.gradle file, ensure that the Java version is specified. It should look something like this:

android {
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
}

Reinstall Android Studio: If you’ve already tried reinstalling, make sure to completely remove any leftover configuration files that might be causing issues. You can do this by deleting the .AndroidStudio directory in your user folder before reinstalling.

Use Java Files: If Kotlin is the default, you can still create Java files manually by right-clicking on your project directory in the Project view, selecting New > Java Class, and then coding your application in Java.

Consult Documentation: Review the official Android documentation for setting up Java projects in Android Studio, as it may provide additional insights or updates.

If you continue to experience issues, consider checking the Android Studio forums or community for similar cases. Good luck!

Best,

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @ahmedmuhmmed8080 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by ghost** [comment]
Hi, see if you have a Java development kit, if not, try downloading it, it might work. If you have this installed, then look at what your default language is in the settings.
